practice/validator/site.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+/**
+ * The Validator takes in the user's input as they type into any given text box
+ * in the form and uses kestroke events to visually que the user that their
+ * input is either valid or not, by filling the field with a yellow background
+ * while invalid.
+ */
+
+/**
+ * Adds the invalid class to the given field (distinguished by a class).
+ *
+ * The invalid class has a yellow background in the site.css file.
+ */
+function makeFieldYellow(inputclass) {
+  $('input.' + inputclass).addClass('invalid');
+}
+
+/**
+ * Removes the invalid class from a given field (distinguished by a class).
+ */
+function removeYellowFromField(inputclass) {
+  $('input.' + inputclass).removeClass('invalid');
+}
+
+/**
+ * Takes in a field (distinguished by a class) and the entry that has been
+ * typed into the textbox of the given field, and returns a boolen representing
+ * whether or not the entry is valid for the given field.
+ */
+function checkIfValid(inputclass, entry) {
+  if (inputclass === 'fullname') {
+    return /[a-z]+\s[a-z]+/i.test(entry);
+  } else if (inputclass === 'birthday') {
+    return /\d{4}\-(\d{2})\-(\d{2})/.test(entry);
+  } else if (inputclass === 'phone-number') {
+    return /(\d{3})\-(\d{3})\-(\d{4})/.test(entry);
+  }
+}
+
+
+/**
+ * Takes in an inputclass, determines it's current entry, checks to see if that
+ * entry is valid, empty, or invalid. If it is valid or empty, it turns off the
+ * yellow background, but if it is invalid, the background becomes yellow.
+ */
+function validateEntry(inputclass) {
+  var entry = $('input.' + inputclass).prop('value');
+  if (checkIfValid(inputclass, entry)) {
+    removeYellowFromField(inputclass);
+  } else if (!entry) {
+    removeYellowFromField(inputclass);
+  } else {
+    makeFieldYellow(inputclass);
+  }
+}
```

Comment:
I like your intuition to remove as much duplicate code as possible, but this structure is very hierarchical and makes it difficult to test any step in isolation. There's no way to ask: Is this string a valid phone number?

The three main steps are

1. Get input value
1. Is valid or empty?
1. Set color

Try to have a main that cleanly does each of those things separately.

Also, this code re-selects elements a ton. In effect, each main could select an input element once, then operate on the selection; that also further decouples what CSS classes are needed from the JS.